media-sound/sinus/patches/sinus-0.1.3.patchset

```diff
@@ -0,0 +1,53 @@
+From f74acf970d4b38e32de661e803e188293400e992 Mon Sep 17 00:00:00 2001
+From: Bach Nguyen <bach5000@gmail.com>
+Date: Thu, 22 Nov 2018 11:31:06 +0000
+Subject: Support gcc2
+
+
+diff --git a/sinus.c b/sinus.c
+index aaab435..34c01a5 100644
+--- a/sinus.c
++++ b/sinus.c
+@@ -235,29 +235,27 @@ void wave_to_little_endian(struct wave_t *wave)
+ #ifdef __LITTLEENDIAN__
+   return;
+ #else
+-  int i;
+-  void *p;
+ 
++  int i = 0;
++  INT16 *p;
+   switch(wave->point_size)
+   {
+     case 1:
+       return;
+     case 2:
+-      p = wave->data;
+-      for(i=0; i<wave->data_size; i+=2)
++      for(p = wave->data; i < wave->data_size; i += 2)
+       {
+-        dprintf((stderr, "host -> le: %04X ->", (UINT16)*((UINT16 *)(p+i))));
+-        (INT16)*((UINT16 *)(p+i)) = HOST_TO_LENDIAN_INT16(*((UINT16 *)(p+i)));
+-        dprint((stderr, " %04X\n", (UINT16)*((UINT16 *)(p+i))));
++        dprintf((stderr, "host -> le: %04X ->", *p));
++        *p = HOST_TO_LENDIAN_INT16(*p);
++        dprint((stderr, " %04X\n", *p));
+       }
+       return;
+     case 4:
+-      p = wave->data;
+-      for(i=0; i<wave->data_size; i+=4)
++      for(INT32 *p = wave->data; i < wave->data_size; i += 2)
```

Comment:
@bach5000 Did you test it for gcc2? I think This will fail.

Reply:
@owenca I did... it works?
Is it not suppose to work?

Reply:
See my [comments](url
https://github.com/haikuports/haikuports/pull/3397#discussion_r235792917) above.

Reply:
@owenca Yes. I just saw your comment :)
Will fix.